src/services/sessionService.ts

```diff
@@ -30,78 +27,58 @@ class Session implements SessionInterface {
 
   private jwkSet;
 
-  constructor(
-    client: AxiosInstance,
-    shortSessionCookieName: string,
-    issuer: string,
-    jwksURI: string,
-    cacheMaxAge: number,
-    projectID: string,
-  ) {
+  constructor(shortSessionCookieName: string, issuer: string, jwksURI: string, cacheMaxAge: number, projectID: string) {
     if (!shortSessionCookieName || !issuer || !jwksURI) {
       throw new Error('Required parameter is empty');
     }
 
-    this.client = client;
     this.issuer = issuer;
     this.cacheMaxAge = cacheMaxAge;
     this.jwkSet = createRemoteJWKSet(new URL(jwksURI), {
       cacheMaxAge: this.cacheMaxAge,
       cooldownDuration: this.cacheMaxAge,
-      headers: {
-        'X-Corbado-ProjectID': projectID,
-      }
+      headers: { 'X-Corbado-ProjectID': projectID },
     });
   }
 
-  public async validateShortSessionValue(shortSession: string): Promise<User> {
+  public async getAndValidateCurrentUser(shortSession: string): Promise<User> {
     Assert.notEmptyString(shortSession, 'shortSession not given');
 
-    if (!shortSession) {
-      return new User(false);
+    if (shortSession.length < MIN_SHORT_SESSION_LENGTH) {
+      throw new JWTError(JWTErrorNames.InvalidShortSession);
     }
 
     try {
       const { payload } = await jwtVerify(shortSession, this.jwkSet, { issuer: this.issuer });
 
-      const { sub, name, email, phoneNumber } = payload as MyJWTPayload;
+      const { userID, fullName, status, explicitWebauthnID } = payload as MyJWTPayload;
 
       if (payload.iss && payload.iss !== this.issuer) {
         this.setIssuerMismatchError(payload.iss);
-        return new User(false);
+        throw new JWTError(JWTErrorNames.InvalidIssuer);
       }
 
-      return new User(true, sub, name, email, phoneNumber);
+      return { userID, fullName, status, explicitWebauthnID };
     } catch (error) {
-      this.setValidationError(error);
-      return new User(false);
-    }
-  }
+      if (error instanceof errors.JWTClaimValidationFailed) {
+        throw new JWTError(JWTErrorNames.JWTClaimValidationFailed);
+      }
 
-  public getLastShortSessionValidationResult(): string {
-    return this.lastShortSessionValidationResult;
-  }
+      if (error instanceof errors.JWTExpired) {
+        throw new JWTError(JWTErrorNames.JWTExpired);
+      }
 
-  public async getCurrentUser(shortSession: string): Promise<User> {
-    if (!shortSession || shortSession.length < MIN_SHORT_SESSION_LENGTH) {
-      return new User(false);
-    }
+      if (error instanceof errors.JWTInvalid) {
+        throw new JWTError(JWTErrorNames.JWTInvalid);
+      }
 
-    const user = await this.validateShortSessionValue(shortSession);
-    return user ?? new User(false);
+      throw error;
+    }
   }
 
   private setIssuerMismatchError(issuer: string): void {
```

Comment:
is this a leftover that we can remove (along with the private lastShortSessionValidationResult)?